src/library/treeitemmodel.cpp

```diff
@@ -193,3 +193,9 @@ TreeItem* TreeItemModel::getItem(const QModelIndex &index) const {
     }
     return m_pRootItem;
 }
+
+void TreeItemModel::triggerRepaint() {
+    //todo: find a nicer way to do this. Works as it should but kind of hackish
```

Comment:
did you try 
emit dataChanged() 
http://doc.qt.io/qt-5/qabstractitemmodel.html#dataChanged
"When reimplementing the setData() function, this signal must be emitted explicitly."

QModelIndex().parent() looks a bit scary can we use at least the real indexes here? 


This function repaints the whole tree, for some extra points, it would be nice to invalidate only the effected items.